File: packages/engine/src/cpus/6502/instructions/tsx.js
import { CycleType } from '..'

/**
 * Transfer Stack pointer to X
 *
 * X = SP
 *
 * @example
 * MODE          SYNTAX        HEX LEN TIM
 * Implied       TSX           $BA  1   2
 */
export function* tsx() {
  this.x = this.sp
  this.zero = this.x === 0
  this.negative = (this.x & 0x80) > 0
  yield CycleType.GET
}

File: packages/engine/src/cpus/6502/instructions/tsx.zig
const std = @import("std");
const CPU = @import("../cpu.zig").CPU;
const Addressing = @import("../opcode.zig").Addressing;

// Transfer Stack pointer to X
//
// X = SP
//
// MODE          SYNTAX        HEX LEN TIM
// Implied       TSX           $BA  1   2

pub fn tsx(self: *CPU, addressing: Addressing) void {
    _ = addressing;
    self.x = self.sp;
    self.z = self.x == 0;
    self.n = (self.x & 0x80) > 0;
}
